tests/components/shelly/test_config_flow.py

```diff
@@ -102,6 +111,33 @@ async def test_form(
     assert len(mock_setup_entry.mock_calls) == 1
 
 
+async def test_form_gen1_custom_port(
+    hass: HomeAssistant,
+    mock_block_device: Mock,
+) -> None:
+    """Test we get the form."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == data_entry_flow.FlowResultType.FORM
+    assert result["errors"] == {}
+
+    with patch(
+        "homeassistant.components.shelly.config_flow.get_info",
+        return_value={"mac": "test-mac", "type": MODEL_1, "gen": 1},
+    ), patch(
+        "homeassistant.components.shelly.config_flow.validate_input",
```

Comment:
You should not patch the code you test itself, you should patch the upstream method. This patch removes 46 lines from being tested. Please patch `BlockDevice.create`

A similar example for patching upstream in this test is the `"homeassistant.components.shelly.config_flow.get_info"` which patch the `get_info` from `aioshelly` so only the upstream code is not being tested here.

Reply:
Done